src/mixins/raster-layer-cross-section-terrain-mixin.js

```diff
@@ -0,0 +1,325 @@
+import assert from "assert"
+import { parser } from "../utils/utils"
+import { AABox2d } from "@heavyai/draw/dist/draw"
+import { lastFilteredSize } from "../core/core-async"
+import { createRasterLayerGetterSetter } from "../utils/utils-vega"
+import { isValidPostFilter } from "./raster-layer-point-mixin"
+import RasterLayerContext from "./render-vega-lite/RasterLayerContext"
+import {
+  ColorChannelDescriptor,
+  GeographicChannelDescriptor,
+  OpacityChannelDescriptor,
+  PositionChannelDescriptor,
+  PropLocation
+} from "./render-vega-lite/PropDescriptor/CommonChannelDescriptors"
+import StringPropDescriptor from "./render-vega-lite/PropDescriptor/BaseTypes/StringPropDescriptor"
+import NumericPropDescriptor from "./render-vega-lite/PropDescriptor/BaseTypes/NumericPropDescriptor"
+import { is_zero_to_one } from "./render-vega-lite/PropDescriptor/BaseTypes/NumericPropValidators"
+import CrossSectionTerrainConfigDefinitionObject from "./render-vega-lite/Definitions/Mark/CrossSectionTerrainConfigDefinitionObject"
+import { materializePropDescriptors } from "./render-vega-lite/RenderVegaLite"
+
+// only used for jsdoc
+// eslint-disable-next-line no-unused-vars
+import VegaPropertyOutputState from "./render-vega-lite/VegaPropertyOutputState"
+
+// eslint-disable-next-line no-warning-comments
+// TODO(croot): this seems like it is used in at least one other layer mixin. Make into a utility somewhere
+function create_post_filter_transform(post_filters) {
+  const post_filter =
+    post_filters && Array.isArray(post_filters) ? post_filters[0] : null // may change to map when we have more than one postFilter
+  if (post_filter && isValidPostFilter(post_filter)) {
+    return {
+      type: "postFilter",
+      table: post_filter.table || null,
+      aggType: post_filter.aggType,
+      custom: post_filter.custom,
+      fields: [post_filter.value],
+      ops: post_filter.operator,
+      min: post_filter.min,
+      max: post_filter.max
+    }
+  }
+  return null
+}
+
+export default function rasterLayerCrossSectionTerrainMixin(_layer) {
+  let state = null
+
+  /**
+   * @type {VegaPropertyOutputState}
+   */
+  let _vega_property_output_state = null
+
+  _layer.setState = function(setter) {
+    if (typeof setter === "function") {
+      state = setter(state)
+    } else {
+      state = setter
+    }
+    return _layer
+  }
+
+  _layer.getState = function() {
+    return state
+  }
+
+  // NOTE: as of 11/14/22 the "getTransforms" method that is found in most of the other raster layer mixin classes
+  // only seems to be called from immerse via the buildRasterExportSql() method in raster-sql.ts
+  //
+  // It is not currently believed that Mesh2d marks need to support export, so the "getTransforms" method is not
+  // going to be included
+
+  const prop_descriptors = new Map()
```

Comment:
Samesies (camel case), if you have it in you to rename things throughout the new mixin 😅

Reply:
I'm fine with function names (like `create_post_filter_transforms`), but every variable throughout these new mixins and his sql library is written in c++ style, I don't have the energy to change them all 😆 
The mix of both styles is bothering me though, so may go back through and refactor when I have the time